server/routes/makeAReferral/completion-deadline/completionDeadlineForm.ts

```diff
@@ -1,20 +1,86 @@
 import { Request } from 'express'
-import DraftReferral from '../../../models/draftReferral'
+import * as ExpressValidator from 'express-validator'
 import errorMessages from '../../../utils/errorMessages'
 import CalendarDayInput from '../../../utils/forms/inputs/calendarDayInput'
 import { FormData } from '../../../utils/forms/formData'
+import { ReferralDetailsFormUpdate } from '../../../models/referralUpdate'
+import { FormValidationError } from '../../../utils/formValidationError'
+import FormUtils from '../../../utils/formUtils'
+import { FormValidationResult } from '../../../utils/forms/formValidationResult'
 
 export default class CompletionDeadlineForm {
-  constructor(private readonly request: Request) {}
+  constructor(private readonly request: Request, private readonly isSentReferral: boolean) {}
 
-  async data(): Promise<FormData<Partial<DraftReferral>>> {
+  static readonly completionDeadlineFieldId = 'completion-deadline'
+
+  static readonly reasonForChangeFieldId = 'reason-for-change'
+
+  async data(): Promise<FormData<ReferralDetailsFormUpdate>> {
     const input = new CalendarDayInput(this.request, 'completion-deadline', errorMessages.completionDeadline)
```

Comment:
nit on naming (not a blocker) - would be good to label this var as the date input